### DISCUSSION TITLE
Toggle component that supports text inside the toggle

### DISCUSSION THREAD

**Discussion by mcullifer** [saadeghi/daisyui#3288] [Ideas / feature requests]
I think it would be nice for the existing [Toggle component](https://daisyui.com/components/toggle) to have a variant that allows text inside the switch. The circle inside the toggle would be the width of the text. Toggling would transition its position to the other side and transition its width. Basically a similar little animation that the current Toggle component already has. Not sure how complicated this is to implement but it seems like it could be done with a similar structure to the [Swap component](https://daisyui.com/components/swap)? I guess with the different toggle types (toggle-primary, toggle-secondary, etc.) the `toggle-on` and `toggle-off` classes would have to apply `text-primary-content`, `text-secondary-content`, etc accordingly.

```html
<label class="toggle">
  <input type="checkbox" />
  <div class="toggle-on">ON</div>
  <div class="toggle-off">OFF</div>
</label>
```

This [medium article](https://hudecz.medium.com/how-to-create-a-pure-css-toggle-button-2fcc955a8984) has a pretty good example and gets the idea across. Could be more refined though like having `select-none`/`user-select: none;` seems pretty sensible because it's kinda gross accidentally double clicking that and it ends up highlighting the text.

Another example is the toggle switch uber eats has
<img width="185" alt="image" src="https://github.com/user-attachments/assets/83bab4cc-3834-4a79-ade2-3f0e4103bd66">
<img width="190" alt="image" src="https://github.com/user-attachments/assets/36500bde-0b3b-48d1-985e-ca24f8d7b16b">

Let me know if this is already possible because I'd love to have something like this in a project I'm using DaisyUI for!

**Comment by saadeghi** [comment]
This will be in daisyUI 5 (coming soon)!
